crates/turborepo-lib/src/run/builder.rs

```diff
@@ -425,6 +448,12 @@ impl RunBuilder {
         }))
         .build()?;
 
+        // If we have an initial task, we prune out the engine to only
+        // tasks that are reachable from that initial task.
+        if let Some(entrypoint_package) = &self.entrypoint_package {
```

Comment:
I think this will eventually need to be a set of packages, especially with debouncing and waiting for current runs to finish being cancelled or completed